bundles/org.openhab.ui/web/src/pages/settings/rules/script/blockly-editor.vue

```diff
@@ -328,156 +266,388 @@
+                <field name="TEXT">abc</field>
+              </shadow>
+            </value>
+          </block>
+          <block type="oh_print">
+            <value name="message">
+              <shadow type="text">
+                <field name="TEXT">abc</field>
+              </shadow>
+            </value>
+          </block>
+        </category>
       </category>
-
       <sep />
-      <category name="Variables" colour="%{BKY_VARIABLES_HUE}" custom="VARIABLE" />
-      <category name="Functions" colour="%{BKY_PROCEDURES_HUE}" custom="PROCEDURE" />
+      <category
+        name="Variables"
+        colour="%{BKY_VARIABLES_HUE}"
+        custom="VARIABLE" />
+      <category
+        name="Functions"
+        colour="%{BKY_PROCEDURES_HUE}"
+        custom="PROCEDURE" />
     </xml>
   </div>
 </template>
 
 <style lang="stylus">
-.blockly-editor
-  display block
-  top calc(100%)
-  height calc(100%)
-  width 100%
-  .blockly
-    height 100%
-    .blocklyMainBackground
-      stroke inherit
-.blocklyDropDownDiv
-  z-index 9000
+.blockly-editor {
+  display: block;
+  top: calc(100%);
+  height: calc(100%);
+  width: 100%;
+
+  .blockly {
+    height: 100%;
+
+    .blocklyMainBackground {
+      stroke: inherit;
+    }
+  }
+}
+
+.blocklyDropDownDiv {
+  z-index: 9000;
+}
 </style>
 
 <script>
 import Blockly from 'blockly'
+import '@blockly/block-plus-minus'
+
 import Vue from 'vue'
 
 import defineOHBlocks from '@/assets/definitions/blockly/ohblocks'
+import defineOHBlocksTimers from '@/assets/definitions/blockly/ohblocks_timers'
+import defineOHBlocksEphemeris from '@/assets/definitions/blockly/ohblocks_ephemeris'
+import defineOHBlocksAudio from '@/assets/definitions/blockly/ohblocks_audio'
+import defineOHBlocksBusEvents from '@/assets/definitions/blockly/ohblocks_busevents'
+import defineOHBlocksLogging from '@/assets/definitions/blockly/ohblocks_logging'
 
-Vue.config.ignoredElements = ['field', 'block', 'category', 'xml', 'mutation', 'value', 'sep']
+Vue.config.ignoredElements = [
+  'field',
+  'block',
+  'category',
+  'xml',
+  'mutation',
+  'value',
+  'sep'
+]
 
 export default {
   props: ['blocks'],
   data () {
     return {
       workspace: null,
-      loading: true
+      sinks: [],
+      voices: [],
+      scripts: [],
+      rules: [],
+      addons: [],
+      loading: true,
+      ready: false
     }
   },
-  mounted () {
-    defineOHBlocks(this.$f7)
-    this.workspace = Blockly.inject(this.$refs.blocklyEditor, {
-      toolbox: this.$refs.toolbox,
-      horizontalLayout: !this.$device.desktop,
-      theme: (this.$f7.data.themeOptions.dark === 'dark') ? 'dark' : undefined,
-      trashcan: false
-    })
-    const xml = Blockly.Xml.textToDom(this.blocks)
-    Blockly.Xml.domToWorkspace(xml, this.workspace)
-
-    this.workspace.addChangeListener(this.onChange)
-  },
-  beforeDestroy () {
-    this.workspace.removeChangeListener(this.onChange)
+  created () {
+    this.getAltData()
+    this.getAddons()
```

Comment:
What is this for?

Reply:
This is used to retrieve the information from the backend to have the voices, scripts, rules and addons available to provide them as lookups in the dropdowns.